src/audio/mixer.c

```diff
@@ -253,41 +261,47 @@ static int mixer_copy(struct comp_dev *dev)
 	if (num_mix_sources == 0)
 		return 0;
 
-	/* make sure no sources have underruns */
-	for (i = 0; i < num_mix_sources; i++) {
+	/* check for overrun */
+	if (sink->free == 0) {
```

Comment:
@tlauda this can be if (!sink->free). But why dont we check is sink->free >= source_bytes here?

Reply:
@ranj063 The intention of the patch is to remove the need of limiting copy to period_size. We should copy as much data as we possible can.

Reply:
@tlauda that intent needs to be in a comment or in the commit message :)